jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/util/HumanTaskHandlerHelper.java

```diff
@@ -201,6 +220,48 @@ public static Deadlines setDeadlines(Map<String, Object> parameters, List<Organi
 		return notifications;
 	}
 
+	protected static int getDeadlineRepeatLimit(String deadlineStr) {
+		String[] repetableParts = DateTimeUtils.parseISORepeatable(deadlineStr);
+		String repeatLimit = repetableParts[0];
+		if (!repeatLimit.isEmpty()) {
+			return Integer.parseInt(repeatLimit);
+		} else {
+			return -1;
+		}
+	}
+
+	protected static Date getDeadlineDurationDate(String durationStr, Date fromDate) {
+		// handles iso repetable and both period and date notation
+		try {
+
+			long fromMilli;
+			if(fromDate != null) {
+				fromMilli = fromDate.getTime();
+			} else {
+				fromMilli = System.currentTimeMillis();
+			}
+			if (DateTimeUtils.isRepeatable(durationStr)) {
+				String[] repetableParts = DateTimeUtils.parseISORepeatable(durationStr);
+				String tempDateTimeStr = repetableParts[2];
+
+				if (DateTimeUtils.isPeriod((tempDateTimeStr))) {
+					return new Date(fromMilli + Duration.parse(tempDateTimeStr).toMillis());
+				} else {
+					return new Date(fromMilli+ DateTimeUtils.parseDateAsDuration(tempDateTimeStr));
+				}
+			} else {
+				if (DateTimeUtils.isPeriod((durationStr))) {
+					return new Date(fromMilli + Duration.parse(durationStr).toMillis());
+				} else {
+					return new Date(fromMilli + DateTimeUtils.parseDateAsDuration(durationStr));
+				}
```

Comment:
Just a note that with timers we also support notation 3s###4s like delay###period, but not sure we want this here.

Reply:
No, we don’t want that here, it’s legacy format that should no longer be used